jwst/assign_wcs/nircam.py

```diff
@@ -48,11 +48,15 @@ def imaging_distortion(input_model, reference_files):
 
 
 exp_type2transform = {'nrc_image': imaging,
-                      'nrc_slitless': not_implemented_mode, #WFSS mode
-                      'nrc_tacq': not_implemented_mode,#       ?? distortion
-                      'nrc_coron': not_implemented_mode,#    ?? distortion
-                      'nrc_focus': not_implemented_mode,#       ?? distortion
-                      'nrc_tss': not_implemented_mode,# custom SOSS like mode TBC
-                      'nrc_tsi': not_implemented_mode,# custom soss like mode (TBC
-                      'nrc_led': not_implemented_mode,#    ?? WFSS mode
+                      'nrc_grism': not_implemented_mode,
+                      'nrc_tacq': imaging,
+                      'nrc_taconfirm': imaging,
+                      'nrc_coron': not_implemented_mode,
+                      'nrc_focus': imaging,
+                      'nrc_tsimage': imaging,
+                      'nrc_tsgrism': not_implemented_mode,
+                      'nrc_led': not_implemented_mode,
+                      'nrc_dark': not_implemented_mode,
+                      'nrc_flat': not_implemented_mode,
+                      'nrc_wfsc': not_implemented_mode,
```

Comment:
The only outlier I see here is nrc_wfsc. I don't have that in any of my lists of NIRCam exp_types. I think wavefront sensing and control (WFSC) exposure are only identified via another flag in their meta data, not by exp_type (probably have regular NRC_IMAGE). For the assignments, all the imaging entires look good. I'm not sure if NRC_CORON would also use imaging. Appropriate to have flat, led, and dark not implemented. Eventually NRC_GRISM and NRC_TSGRISM will be implemented, but they haven't been yet.